relay-server/src/endpoints/common.rs

```diff
@@ -347,13 +349,6 @@ pub async fn handle_envelope(
         )
     }
 
-    // TODO(jjbayer): Move this check to spool impl
-    if state.memory_checker().check_memory().is_exceeded() {
```

Comment:
Final thought: We should restore this, because the `has_capacity` check only runs for spool V2.

Reply:
Oh good point, thanks for pointing that out!